toqito/nonlocal_games/tests/test_extended_nonlocal_game.py

```diff
@@ -186,3 +189,267 @@ def test_chsh_commuting_value_upper_bound(self):
+        #    self.skipTest(f"Test not configured for platform: {current_platform}")
+        #    expected_ns_value = (3 + np.sqrt(5)) / 6.0
+        
+        # self.assertAlmostEqual(
+        #    ent_lb,
+        #    2/3.0,
+        #    delta=1e-4,
+        #    msg=f"Value {ent_lb} does not match expected {expected_ns_value} on {current_platform}"
+        # )
+
+    def test_quantum_lb_alice_opt_fails_status(self):
+        """Test quantum_value_lower_bound when Alice's optimization fails (bad status)."""
+        prob_mat, pred_mat = self.bb84_extended_nonlocal_game()  # Use any valid game
+        game = ExtendedNonlocalGame(prob_mat, pred_mat)
+
+        # Mock __optimize_alice to simulate a failure status
+        mock_problem_alice = mock.Mock(spec=cvxpy.Problem)
+        mock_problem_alice.status = cvxpy.SOLVER_ERROR  # Or any non-OPTIMAL status
+        mock_problem_alice.value = None
+
+        with mock.patch.object(game, "_ExtendedNonlocalGame__optimize_alice", return_value=(None, mock_problem_alice)):
+            # Expecting 0.0 as the fallback when current_best_lower_bound is -inf
+            res = game.quantum_value_lower_bound(iters=5, seed=0)
+            self.assertEqual(res, 0.0)
+
+    def test_quantum_lb_alice_opt_fails_value_none(self):
+        """Test quantum_value_lower_bound when Alice's optimization returns value=None."""
+        prob_mat, pred_mat = self.bb84_extended_nonlocal_game()
+        game = ExtendedNonlocalGame(prob_mat, pred_mat)
+
+        mock_problem_alice_val_none = mock.Mock(spec=cvxpy.Problem)
+        mock_problem_alice_val_none.status = cvxpy.OPTIMAL_INACCURATE  # Status is acceptable
+        mock_problem_alice_val_none.value = None  # But value is None
+
+        with mock.patch.object(
+            game, "_ExtendedNonlocalGame__optimize_alice", return_value=(None, mock_problem_alice_val_none)
+        ):
+            res = game.quantum_value_lower_bound(iters=5, seed=0)
+            self.assertEqual(res, 0.0)
+
+    def test_quantum_lb_bob_opt_fails_status(self):
+        """Test quantum_value_lower_bound when Bob's optimization fails (bad status)."""
+        prob_mat, pred_mat = self.bb84_extended_nonlocal_game()
+        game = ExtendedNonlocalGame(prob_mat, pred_mat)
+
+        # Alice's step needs to succeed and return valid rho variables
+        mock_rho_vars = defaultdict(
+            lambda: mock.Mock(spec=cvxpy.Variable, value=np.array([[0.5, 0], [0, 0.5]]))
+        )  # Dummy valid value
+
+        mock_problem_alice_ok = mock.Mock(spec=cvxpy.Problem)
+        mock_problem_alice_ok.status = cvxpy.OPTIMAL
+        mock_problem_alice_ok.value = 0.5  # Dummy value
+
+        mock_problem_bob_fail = mock.Mock(spec=cvxpy.Problem)
+        mock_problem_bob_fail.status = cvxpy.SOLVER_ERROR
+        mock_problem_bob_fail.value = None
+
+        with mock.patch.object(
+            game, "_ExtendedNonlocalGame__optimize_alice", return_value=(mock_rho_vars, mock_problem_alice_ok)
```

Comment:
Why is there an underscore for `_ExtendedNonlocalGame__optimize_alice`?

Reply:
its just for extra check, because optimizer depends on hardware architecture